src/main/java/org/jabref/logic/shared/OracleProcessor.java

```diff
@@ -118,7 +118,8 @@ protected void insertIntoEntryTable(List<BibEntry> entries) {
             }
             insertEntryQuery.append(" SELECT * FROM DUAL");
             LOGGER.info(insertEntryQuery.toString());
-            try (PreparedStatement preparedEntryStatement = connection.prepareStatement(insertEntryQuery.toString())) {
+            try (PreparedStatement preparedEntryStatement = connection.prepareStatement(insertEntryQuery.toString(),
+                    new String[]{"SHARED_ID"})) {
```

Comment:
I am not sure, but this looks odd to me with the String[]

Reply:
This just indicates the column where we want to put the auto-generated keys. See https://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html#prepareStatement(java.lang.String,%20int[])

Reply:
I looked into Stack Overflow and the Oracle JDBC driver source code and it looks like the Oracle JDBC doesn't support `getGeneratedKeys` on `INSERT ALL` statements. So I'll have to find another way of getting the shared IDs.

Reply:
OMG. This sounds like we should really drop Oracle support. There is no need for Oracle in 2020, is it? - Postgres is the way to go, isn't it?

We should IMHO also drop MySQL support as it does not offer automatic updates on changes on the server.